uber/site_sections/registration.py

```diff
@@ -318,12 +319,18 @@ def check_in(self, session, id, age_group, group, badge_num=None, message=''):
                     message = 'That badge number already belongs to ' + maybe_dupe.first().full_name
 
             if group:
-                session.match_to_group(attendee, session.group(group))
+                g = sesson.group(group)
+                if g.amount_unpaid:
+                    message = 'That group has an outstanding balance of ${}'.format(g.amount_unpaid)
+                else:
+                    session.match_to_group(attendee, g)
             elif attendee.paid == c.PAID_BY_GROUP and not attendee.group:
                 message = 'You must select a group for this attendee.'
 
-            if attendee.badge_status != c.COMPLETED_STATUS:
-                message = 'This badge is {0} and cannot be checked in.'.format(attendee.badge_status_label)
+            if not message:
+                attendee._status_adjustments()
+                if attendee.badge_status != c.COMPLETED_STATUS:
+                    message = 'This badge is {} and cannot be checked in.'.format(attendee.badge_status_label)
```

Comment:
Side note for @kitsuta: in the past, we've generally just kept placeholder registrations in the system and even allowed them to check in, so long as they have an age group and an emergency contact number.  This badge status code which I believe you write before MagClassic prevents that from happening; in such cases, an admin would need to bring up the admin form and fill out the rest of their information.

Is that what we want?  Or should we instead confirm that all required information is present and then de-select the badge as a placeholder?  Your call.